app/frontend/actions/wallet.ts

```diff
@@ -56,9 +42,37 @@ const accountsIncludeStakingAddresses = (
 export default (store: Store) => {
   const {loadingAction} = loadingActions(store)
   const {setError} = errorActions(store)
-  const {setState} = store
+  const {getState, setState} = store
   const {setWalletOperationStatusType} = commonActions(store)
 
+  const loadAsyncWalletData = async (): Promise<void> => {
+    const asyncFetchAndUpdate = async <T extends Partial<State>>(
+      fetch: () => Promise<T>,
```

Comment:
fetchFn would imho be better, to not confuse it with native "fetch" function in js

Reply:
Refactored